kpi/serializers/v2/open_rosa.py

```diff
@@ -0,0 +1,99 @@
+    descriptionText = serializers.SerializerMethodField('get_description')
+    downloadUrl = serializers.SerializerMethodField('get_download_url')
+    manifestUrl = serializers.SerializerMethodField('get_manifest_url')
+
+    def get_description(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.descriptionText
+
+    def get_download_url(self, obj):
+        self.__validate_object_inheritance(obj)
+        request = self.context['request']
+        return obj.get_download_url(request)
+
+    def get_form_id(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.form_id
+
+    def get_hash(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.hash
+
+    def get_manifest_url(self, obj):
+        self.__validate_object_inheritance(obj)
+        request = self.context['request']
+        return obj.get_manifest_url(request)
+
+    def get_name(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.name
+
+    def __validate_object_inheritance(
+        self, obj: OpenRosaFormListInterface
+    ) -> bool:
+        """
+        Validates if object inherits from `AbstractOpenRosaFormListModel`.
+        It helps to catch upstream missing properties and methods when
+        rendering data
+        """
+        # Use private variable to test validation only once per instantiation
+        assert issubclass(obj.__class__, OpenRosaFormListInterface)
+
+
+class ManifestSerializer(serializers.Serializer):
+    """
+    This serializer is model-agnostic. The list of objects passed to the
+    serializer must inherit from `kpi.interfaces.open_rosa.OpenRosaManifestInterface`  # noqa
+    to be sure expected methods and properties are defined.
+    """
+    # The PEP-8 naming convention is broken on purpose
+    # Open Rosa XML uses CamelCase
+    # See https://docs.getodk.org/openrosa-form-list/#the-manifest-document
+    filename = serializers.SerializerMethodField()
+    hash = serializers.SerializerMethodField()
+    downloadUrl = serializers.SerializerMethodField('get_download_url')
+
+    def get_download_url(self, obj):
+        self.__validate_object_inheritance(obj)
+        request = self.context['request']
+        return obj.get_download_url(request)
+
+    def get_filename(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.filename
+
+    def get_hash(self, obj):
+        self.__validate_object_inheritance(obj)
+        return obj.hash
+
+    def __validate_object_inheritance(
+        self, obj: OpenRosaManifestInterface
+    ) -> bool:
+        """
+        Validates if object inherits from `OpenRosaFormModelInterface`.
+        It helps to catch upstream missing properties and methods when 
+        rendering data
+        """
+        # Use private variable to test validation only once per instantiation
+        assert issubclass(obj.__class__, OpenRosaManifestInterface)
```

Comment:
how about `isinstance()`
see also https://github.com/kobotoolbox/kpi/pull/2537/files/?file-filters%5B%5D=.py#diff-fe9e5cb333b07029faea363451f9dac43c5eaba656cd1027174b3f249c8f2b20R61

Reply:
Replaced

Reply:
would be better as `issubclass()` given what you're doing (see [previous](https://github.com/kobotoolbox/kpi/pull/2537#discussion_r644087030) comment)